fairscale/nn/misc/grad_bucket.py

```diff
@@ -16,6 +16,7 @@ class GradBucket:
     def __init__(self, size: int, dtype: torch.dtype, device: torch.device, destination: int) -> None:
         self._max_size = size
         self._params: List[torch.Tensor] = []
+        self._param_ids: List[int] = []
```

Comment:
small extension to fool proof these: forbid the same params being checked in twice

Reply:
Just curious if you have seen this happen. Can you have grads checked in twice?

Reply:
never say never :) Some models share weights in different places, so yes I think this could easily happen, I was trying to be cautious